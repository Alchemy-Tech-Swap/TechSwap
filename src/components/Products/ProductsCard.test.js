import ProductCard from './ProductsCard';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test('Should render Product Card', async () => {
  const { container } = render(
    <MemoryRouter>
      <ProductCard
        products={[
          {
            category: 'Computer',
            condition: 'Used',
            created_at: '2022-01-20T18:38:25.099108+00:00',
            description: 'Won it in a raffle, who wants it!?',
            id: 26,
            image:
              'https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/desk.jpg',
            price: 100,
            sold: null,
            title: 'Standing Desk',
            user_id: 25,
          },
        ]}
      />
    </MemoryRouter>
  );
  await screen.findByText('Standing Desk');
  expect(container).toMatchSnapshot();
});
