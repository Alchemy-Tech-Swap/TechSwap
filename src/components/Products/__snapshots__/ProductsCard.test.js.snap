// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Product Card 1`] = `
<div>
  <div>
    <h1
      class="productListTitle"
    >
      <img
        alt="Techswap2"
        class="techlogo"
        src="tswapbanner.png"
      />
      Welcome To The TechSwap
    </h1>
    <div
      class="productLists"
    >
      <a
        class="productLinks"
        href="/products/26"
      >
        <img
          class="productImage"
          src="https://epowwlsvajvjgzuvqqfe.supabase.co/storage/v1/object/public/product-image/public/desk.jpg"
        />
        <h1
          class="productTitle"
        >
          Standing Desk
        </h1>
        <p
          class="productCondition"
        >
          Used
        </p>
        <p
          class="productPrice"
        >
          $
          100
        </p>
      </a>
    </div>
  </div>
</div>
`;
